### ISSUE TITLE
linter: oxc(no-async-endpoint-handlers) reports Fastify endpoints

### ISSUE THREAD

**Issue by mootoday** [oxc-project/oxc#9802]
### What version of Oxlint are you using?

0.16.0

### What command did you run?

`pnpm dlx oxlint@latest -D suspicious .`

### What does your `.oxlintrc.json` config file look like?

None

### What happened?

Fastify (https://fastify.dev/) async endpoints are considered Express async endpoints.

This is what a Fastify server looks like:

```js
import Fastify from 'fastify'

const fastify = Fastify({
  logger: true
})

fastify.get('/', async (request, reply) => {
  return { hello: 'world' }
})

const start = async () => {
  try {
    await fastify.listen({ port: 3000 })
  } catch (err) {
    fastify.log.error(err)
    process.exit(1)
  }
}
start()
```

Not a blocking issue as I can disable the `oxc(no-async-endpoint-handlers)` rule.

**Comment by nrayburn-tech** [comment]
Looking at the code for the rule, it seems pretty specific to Express < version 5. Other servers like Fastify, Koa, etc. may not have the same limitations.

Probably the best resolution for this is to rename the rule to be Express specific. Users would still need to enable/disable it based on which version of Express they are using. However, it would avoid flagging for other frameworks like Express that do support async handlers.

**Comment by Boshen** [comment]
This rule is intended for express https://oxc.rs/docs/guide/usage/linter/rules/oxc/no-async-endpoint-handlers.html

You may turn it off if you are not using express.
